src/test/java/reposense/model/RepoLocationTest.java

```diff
@@ -1,59 +1,182 @@
+        assertValidLocation("https://github.com/reposense",
+                "reposense", "", EXPECTED_DOMAIN_NAME);
+        assertValidLocation("https://github.com/reposense/RepoSense",
+                "RepoSense", "reposense", EXPECTED_DOMAIN_NAME);
+
+        // Test against other types of URL protocols that are valid for git clone
+        assertValidLocation("ssh://git@github.com/path/to/repo.git/",
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION, EXPECTED_DOMAIN_NAME);
+        assertValidLocation("git://github.com/path/to/repo.git",
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION, EXPECTED_DOMAIN_NAME);
+        assertValidLocation("https://localhost:9000/path/to/repo.git",
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION, EXPECTED_UNRECOGNISED_DOMAIN_NAME);
+
+        // Test against the conventional ssh protocol used for GitHub, e.g. git@github.com:reposense/RepoSense.git
+        assertValidLocation("repo@organization.com:path/to/repo.git/",
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION, EXPECTED_UNRECOGNISED_DOMAIN_NAME);
+        assertValidLocation("git@github.com:reposense/RepoSense.git",
+                "RepoSense", "reposense", EXPECTED_DOMAIN_NAME);
     }
 
     @Test
-    public void repoLocation_repoUrlWithASpecifiedBranch_throwsInvalidLocationException() {
-        // reject both repos with and without .git
-        assertInvalidLocation("https://github.com/reposense/testrepo-Beta/tree/add-config-json");
-        assertInvalidLocation("https://github.com/reposense/testrepo-Beta.git/tree/add-config-json");
+    public void repoLocation_parseNormalizableRepoLocations_success() throws Exception {
+        assertValidLocation("https://github.com/reposense/redundant/directories/../../RepoSense.git",
+                "RepoSense", "reposense", EXPECTED_DOMAIN_NAME);
+        assertValidLocation("/path/with/redundant/directories/../.././../to/repo",
+                EXPECTED_REPO_NAME, "-" + EXPECTED_ORGANIZATION, EXPECTED_UNRECOGNISED_DOMAIN_NAME);
+
+        // Tests if there is an additional '../' it is not consumed by accident
+        assertValidLocation("path/with/redundant/directories/../../../../../to/repo",
+                EXPECTED_REPO_NAME, "..-to", EXPECTED_UNRECOGNISED_DOMAIN_NAME);
     }
 
     @Test
-    public void repoLocationParser_parseEmptyString_success() throws Exception {
-        RepoLocation repoLocation = new RepoLocation("");
+    public void repoLocation_parseInvalidRemoteRepo_throwsInvalidLocationException() throws Exception {
+        // Invalid URL protocol
+        assertInvalidLocation("ttp://github.com/reposense.RepoSense.git");
+        assertInvalidLocation("not-valid-protocol://abc.com/reposense/RepoSense.git");
+        // URL contains illegal characters
+        assertInvalidLocation("https://github.com/contains-illegal-chars/^\\/");
     }
 
     /**
      * Compares the information parsed by the RepoLocation model with the expected information
```

Comment:
I think `assertValidLocation` and `assertInvalidLocation` may need to be renamed because some unrecognised domains are considered "valid" although another developer may think that these unrecognised domains should be considered invalid instead.